### DISCUSSION TITLE
Bugs General

### DISCUSSION THREAD

**Discussion by pavlobu** [pavlobu/deskreen#68] [Bugs]
If you saw or experienced a bug while **using** Deskreen. Drop it here. Please include Deskreen version!
If you saw a bug and know your dev environment, and how to reproduce it, please consider opening a new Issue labeled as **Bug** and provide full details.

**Comment by kernel0** [comment]
I use deskreen 1.0.3 with dummy plug adapter. Good connection but I can't watch my cursor on my third screen created by deskreen. Anything else is good.

**Reply by kernel0** [reply]
my environment : 
- desktop pc windows 10 pro 20H2 (build : 19042.746, Windows Feature Experience Pack 120.2212.551.0)
- second display : physical device (LG )
- third display(deskreen) : I tested with ipad pro 10.5 safari and macbook pro 2013 safari.

I experienced the bug on deskreen 1.0.0 and 1.0.2, too.

**Reply by kernel0** [reply]
Also I tested two another configuration.
1.without dummy plug,  first screen entire sharing -> ok. I watched the cursor.
2.without dummy plug,  second screen (physical device) entire sharing -> no, I can't watch that.

**Reply by kernel0** [reply]
Last I tested with dummy plug and application sharing, watched cursor only but Application window is entire black (maybe not painted)

**Reply by pavlobu** [reply]
Try latest version

**Reply by kernel0** [reply]
1.0.3 too..

**Reply by pavlobu** [reply]
Have no clue, 
Try to go through this:
https://github.com/pavlobu/deskreen/issues/45

**Comment by kernel0** [comment]
reproduce it! (cursor disappear issue)

No.3 monitor is deskreen 

working
![image](https://user-images.githubusercontent.com/1070578/106587550-1ad6a200-658d-11eb-9ec2-8aea04bff983.png)

not working.
![image](https://user-images.githubusercontent.com/1070578/106587619-2e820880-658d-11eb-8d24-791cf56a4e72.png)

not working
![image](https://user-images.githubusercontent.com/1070578/106587802-6ab56900-658d-11eb-819f-5984a12469e9.png)

No.1 monitor resolution is 3440 * 1440 (primary monitor)
No2. monitor resolution is 1440 * 2560

maybe if x size is over (3440+1440) , not working
and if y size is over 1440, not working.

**Reply by pavlobu** [reply]
Did you read #45 ? Try doing this in chromium browser and test with a link in that issue (webrtc examples) if chromium also behaves like that, then it is a problem of chromium, let's wait when they will fix it and I'll update electron version to fix your problem later when it is ready

**Reply by pavlobu** [reply]
https://github.com/pavlobu/deskreen/issues/45#issuecomment-767650890

**Reply by pavlobu** [reply]
I have already fixed a problem like this by just updating chromium version, so if its a chromium issue, we can wait. Or you can [submit this bug to chromium repo](https://github.com/chromium/chromium) -> they fix it -> electron updates chromium engine -> we update Deskreen electron version and it will work finally

**Comment by pavlobu** [comment]
I noticed performance issues on Windows 10 when dragging Deskreen window

Will try to fix it soon

**Comment by easyt0re** [comment]
I'm using Win10 with version 1.0.4. The software just crashes all the time. I suspect it's because my laptop is too old. So I was wondering if some info on the "minimal requirement" of the hardware can be added. 
This post is not really about reporting my crashes but I also feel like it's not a "new feature", so I just drop it here. I read the FAQ somewhere that the software does require some resources to run smoothly.

**Reply by pavlobu** [reply]
ok thank you for writing about it. I will be pushing hard to get rid of those crashes on Win10 . I mostly develop on my mac, and just test if stuff works on my windows 10 virtual machine, and occasionally test on my Win 10 beefy workstation. I have found some lags on Win10 and it is probably that the UI is overloaded somehow with background tasks. Your crashes may be related to this issue:
https://github.com/pavlobu/deskreen/discussions/71#discussioncomment-341268
stay tuned, I will be fixing those issues on win10, it may probably fix your issue as well

and yes, we need to get minimal system requirements. the problem is that my win10 system is not that "minimal".
I think I'll play around with creating a really small VBox instance of win10 and try if Deskreen runs on it, then I'll write min requirement on FAQ.
can you write what is your Win10 setup here?


Thanks in advance,
Paul

**Reply by easyt0re** [reply]
Thanks for the system. It's a really old system. Probably irrelevant. 
It's Lenovo Y40-80 with i7-5500U @ 2.4GHz, 8G RAM, AMD Radeon R9 M275 2GB. 
The version for Windows is probably around 1019. 
I guess this is what you are asking. Like I said, probably not the targeted user machine. 
If this is not enough, just tell me what to do. I can provide screenshots or output from cmd. 

Thanks for the reply. Good work. If this works, it's something I've been looking for.

**Reply by pavlobu** [reply]
Hi, thanks for info. The system parameters look ok and Deskreen should work. Can you try latest but windows portable version and tell if it works?
https://github.com/pavlobu/deskreen/releases/download/v1.0.5/deskreen_1.0.5_Windows_portable.exe

also you can send logs here. just run Deskreen app from cmd and copy paste logs file reference here. Thanks

also make sure antivirus, windows defender are treating Deskreen as exception

**Reply by easyt0re** [reply]
Maybe I should describe my experience more carefully. I can install and run the software, both the 1.0.4 I installed yesterday and the 1.0.5 portable I tried today. I'm using an iPhone 8 with Chrome (or Safari) as my "client/viewer". I almost always get past the "connect" step. Soon after that, my phone shows me "Deskreen Error Dialog: Something wrong happened". 
Yesterday, with 1.0.4, the software started to "Not responding" or simply just froze at this point. Maybe "crashing" was the wrong word to use. I also experienced this "Not responding" directly after opening the software once. It's hard to kill even with the help of the task manager. 
With cmd launching 1.0.5 portable, it's just a clean launch without any output. I just cd to the folder and use the file name to run it. I don't know if there is a more verbose way. This time I got as far as choosing apps to share. Yesterday, I didn't even see this because it's always empty. It wouldn't load the currently available apps. I actually got further to "confirm" today. But after I click that, it will again ask me to "reconnect", the same error as before. 
With 1.0.5 portable, I also try to use another Win10 laptop as the viewer. Similar results happen. I can include a screenshot here. Everything was looking promising until this error shows up. 
![image](https://user-images.githubusercontent.com/9708107/107280789-144a9d80-6a59-11eb-9b55-6204a400276c.png)

In general, the software responds to me quite slowly. After I click close, I have to wait one or two seconds for it to close. That's why I thought my system was probably too old for this. Idk. Maybe it's nothing.

**Reply by pavlobu** [reply]
@easyt0re  Thank you for such a detailed explanation. Indeed I was able to reproduce this issue with a lowest configuration of Windows Virtual machine on which I'm testing Deskreen. 
I will dig further to get rid of this problem in nearest future.

**Reply by pavlobu** [reply]
@easyt0re problem is that I was developing on powerful machine all the time. Now I get useful feedbacks and your feedback also helps. So now I will be working to eliminate these issues in future releases.
I didn't expect Windows will be such a pain and of course I got most of users on Windows ..

**Reply by pavlobu** [reply]
@easyt0re the app is a baby now. It is really hard to build a cross-platform app that works smooth on all platforms and versions of all operating systems from the very first release. Stay tuned for updates and this issue will be fixed, I plan to polish app by the end of Feb, there is a lot of work ahead

**Reply by pavlobu** [reply]
@easyt0re note that this does not happen on the same lowest machine config of linux. I have tested it on Linux Virt. machine. I also successfully ran Deskreen on my old RPI 3B+ it is really weak, but it worked on that board

**Comment by pavlobu** [comment]
Noticed a bug on 1.0.4 . The quality is NOT set for 100% by default when connection starts, will fix it asap.

The goal is to set 100% quality when connection starts. Right now it is set to AUTO

**Reply by pavlobu** [reply]
Fixed in 1.0.5 ⚡️
https://github.com/pavlobu/deskreen/issues/100

https://github.com/pavlobu/deskreen/releases/tag/v1.0.5

**Comment by MaximoMachado** [comment]
I'm on v1.0.5 and everything was working great until today. For whatever reason, I'll connect my Windows desktop to a second screen from a dummy adapter on my linux laptop, put it into fullscreen, and after a couple minutes of use, the windows screen will freeze. The screen freezes for a bit before going black, and once I exit from fullscreen, it provides me with the error, `WebRTC error occurred.` 

Each time I've been trying to reconnect it, the same thing occurs.

**Reply by pavlobu** [reply]
That is really strange. Looks. Like a flaky behavior. Try downgrading version to 1.0.0, or try windows portable 1.0.5 and tell if it helps.

Cheers

**Reply by pavlobu** [reply]
Also maybe reinstalling app may help

**Reply by MaximoMachado** [reply]
I restarted my laptop and so far it appears to be working. I'll report back if it's not just some freak one time occurrence.

Also, thanks so much for making Deskreen, it's solved a problem I was having for far too long. It's great being able to enter the land of two monitors again.

Edit: Just my luck, as soon as I wrote this I got the WebRTC error again. I'll try reinstalling and hopefully that helps out. For reference, I'm on Ubuntu.

**Reply by pavlobu** [reply]
Thanks! Enjoy using it. If you really think it may help others, please share it with them

Try if Appimage works ok on Ubuntu?

**Reply by MaximoMachado** [reply]
For whatever reason, restarting the computer that Deskreen was streaming over to fixed my issue. That probably means Deskreen wasn't at fault then; no clue why restarting the machine that isn't running Deskreen would fix things.

**Comment by ZhangTianrong** [comment]
Using 1.0.5, the app runs extremely slow (I mean navigating within the app window, setting up connection, etc) and chances are that there is no window or screen to select. Have tried previous versions, and similar issues exist. Sometimes when I start as Admin and the windows show up. Also, I have encountered this a few times, when it says something wrong happened on my phone but the allow/disallow dialog already shows up or the stream cannot be opened after I select the screen or window to share but deskreen isn’t aware of this. Furthermore it seems that  screen keyboard can crush the problem sometimes?

**Reply by pavlobu** [reply]
Hi, thank you for providing us with these issues.
I have a clear goal on fixing this problem on windows before code signing is done.
https://github.com/pavlobu/deskreen/discussions/71#discussioncomment-341268
I have a mid 2015 Mac only for development, I run Virtual Machine on Windows to test app there. So I'm pretty tight on resources.
I set a financial goal on opencollective to collect funds for buying a machine that I will use for Deskreen development on Windows and Linux. If you want to help you can contribute.

stay tuned for updates

**Comment by ivanbarlog** [comment]
Using macOS 11.2.1 (20D74) and v1.0.5 of Deskreen I am trying to connect over my home network to my Sony Bravia with it's internal web browser.

I am able to set up the connection but I am stuck at `Waiting for user to seelct source to share from screen sharing device...`

On my macbook I have already set up everything, I can see the `My Device Info` on TV and on my macbook as well -> they match. The Deskreen on macbook say "Success!" but the TV is still waiting.

I've tried this with another device (eg. phone) and it worked. I've also tried the localhost - worked as charm.

The TV and macbook are at the same WiFi network so that shouldn't be a problem. Seems like the browser on my TV is missing some essential functionality? But on macbook everything seems to be connected...

Is there any log I can paste here?

PS: The TV uses [Vewd browser](https://www.vewd.com/), unfortunately I cannot find the version...  

PS2: This is what Deskreen says about the TV:

```
Device Type: computer
Device IP: ...
Device Browser: Opera 36.0.2128.0
Device OS: Linux armv7l
Session ID: ...
```

TV Model Name: KD-55XF7005
Software version: v8.885-1000-2.650-0000 (up to date)

**Reply by pavlobu** [reply]
Hi, I'm sorry but internal web browsers of smart TVs other than Chrome or Firefox are not supported, because they may not have support for WebRTC... :/

If you have an option to install Chrome or Firefox on your TV, then it will work, others have tried that already

Cheers

**Reply by ivanbarlog** [reply]
Thanks! I've read this statement within the FAQs after I've posted this ...


Unfortunately I am unable to install on the TV anything since it is not Androind TV.


Thanks anyway!


BTW: are you planning on supporting any other browsers? I've checked via https://www.webrtc-experiment.com/DetectRTC/ and it seems that there is Support for WebRTC but not for ORTC. I would paste the screenshot here but unfortunately cannot do that :-(

**Comment by hobojoe551** [comment]
So for some reason it just keeps freezing on me. The second I click yes to connect my device the desktop application just freezes. I am running popos currently. I am also using xrandr to make a virtual display, I am not sure if that is what is causing the issue. I'm not exactly sure what troubleshooting steps to go through, or if there is any more information needed. I have tried both the app image and the deb, and have gotten the same result from both.

**Reply by pavlobu** [reply]
Hi, you can try troubleshooting from here:
https://deskreen.com/#faq-4

**Reply by hobojoe551** [reply]
Yuup, those are all steps I took before even reading it. I did however figure out a way to get around the issue. If I maximize the window and the return it to normal size it unfreezes. It lets me continue from there too.

**Reply by pavlobu** [reply]
Cool, there are also play pause buttons in browser ui

**Comment by strannik19** [comment]
ubuntu 20.04. tried using both appimage and deb install. deskreen starts up, but is only listening on ipv6:

```
netstat -an | grep LISTEN | grep 3131                                                                                              
tcp6       0      0 :::3131                 :::*                    LISTEN
```
deskreen itself shows the ipv4 address

**Reply by pavlobu** [reply]
this will be fixed with allowing user to choose which network card to use in future releases.
the feature to allow user to type ip its own known ip address manually will also be added
it is a next thing in my todo list:
https://github.com/pavlobu/deskreen/discussions/71#discussioncomment-321753

stay tuned for updates

Cheers,
Paul

**Reply by strannik19** [reply]
thanks! looking forward to have this working.

**Comment by Joseph-Hui** [comment]
Deskreen 1.0.11 + FluffyDisplay on MacOS Catalina KVM guest. When returning from screen lock, the Deskreen client is blank and does not show anything. Need to disconnect and reconnect again.

**Comment by danielporto** [comment]
Why is the security option grayed out and unavailable in General Settings (Deskreen for mac version 1.0.11) ?

**Comment by danielporto** [comment]
Is there a way to keep the URL? 
I have another old Mac that I use as extra display, but every time it gets disconnected (change network, sleep) I need to type the URL again, it would be nice if the devices could be discovered automatically or at least have a way to make the URL static.

**Comment by metapea** [comment]
- Deskreen keep going on after exiting the program and went over 8gb of ram:
![deskreen](https://user-images.githubusercontent.com/35351174/132202982-d6beaa07-6d72-44f8-9210-57c258c2002c.png)
- It keep flashing on my desktop screen when i turn it on. 
- the stream just stopped and i had to keep making new links for every couples of hours
 minor:
-can you pleases improve the frame rate

**Comment by RBruyn** [comment]
Hi!
I'm having a problem when try to install in Debian the libappindicator was deprecated, how can I fix this ?

**Comment by royyhlee** [comment]
I'm using Opensuse tumbleweed + app image with Deskreen v1.0.12 I was able to figure out the server connection issue with this https://github.com/pavlobu/deskreen/discussions/48#discussioncomment-333047

It is able to connect but my iPad safari keeps showing "WebRTC error occured" is there any way to fix this? I'd love to use this app!

EDIT:
I updated my ipad to the latest version and now it immediately shows "You were not allowed to connect" when I use the QR code

**Comment by smileR-T** [comment]
I used deskreen on X86 Linux system. The device is connected successfully, but sharing the whole screen fails.It is normal to share a separate window. The sharing screen is black and only one mouse is displayed. Please have a look ~

**Reply by jnoon417204** [reply]
@smileR-T I would appreciate any insights if you've managed to overcome this issue. I believe I am experiencing the exact same issue on x86 linux. When sharing apps, I can only share the deskreen application window; moreover, trying to share the entire screen results in a black screen wherein my mouse appears but only if it is within the borders of the deskreen application window.

**Comment by NRSingh007** [comment]
> If you saw or experienced a bug while **using** Deskreen. Drop it here. Please include Deskreen version! If you saw a bug and know your dev environment, and how to reproduce it, please consider opening a new Issue labeled as **Bug** and provide full details.

Recently I downloaded Deskreen Version: 2.0.4 (2.0.4)
and when I tried to share the entire screen with or without a dummy HDMI plug the entire screen showed blank (BLACK)
I am using Ubuntu 22.04
![Screenshot from 2023-03-14 21-07-39](https://user-images.githubusercontent.com/42314977/225054641-631c3b0f-30ba-40c3-b133-6a599c7eee97.png)

![Screenshot from 2023-03-14 21-24-06](https://user-images.githubusercontent.com/42314977/225060524-e19e340e-7a12-4c82-994d-0820cd97d6d2.png)

working fine with app window

![Screenshot from 2023-03-14 21-08-18](https://user-images.githubusercontent.com/42314977/225054889-63215047-c655-4fc6-846e-0ea088a571b8.png)

**Reply by tadeususman** [reply]
Hi There, 

I also experienced it on my ubuntu 22.04.
and this is me browsing the internet, the reason is the wayland problem. by default wayland on ubuntu 22.04 is disabled. then I try to enable it. and I can share the entire screen. 

![Screenshot from 2023-08-24 12-20-08](https://github.com/pavlobu/deskreen/assets/101442771/27bd5414-6ad3-420b-9906-69bbde99b86f)

for enable/disable wayland : https://linuxconfig.org/how-to-enable-disable-wayland-on-ubuntu-22-04-desktop

**Comment by openformoontribe** [comment]
Hi, not sure if this has already been pointed out yet. Downloaded deskreen, but had a WebRTC error. tried different devices. ipad and one of my phones didnt work. But another phone worked. Look for the difference in settings. Turns out to be the VPN. Turn off the VPN and it works.

**Comment by YeXin451664** [comment]
Deskreen如何修改启动的端口号？

**Comment by MGuerrera** [comment]
Using Deskreen 2.0.4 with BetterDisplay (create virtual monitors) on macOs Sonoma, it happens that when the screen saver starts due to inactivity, the virtual monitor on the browser where Deskreen is streaming freezes and turns black. At that point every time I have to close the browser tab and connect Deskreen again.

**Comment by Panashe-Tech** [comment]
Anyone who knows how to fix this issue. Deskreen cannot detect Wifi or Lan connection after upgrading to Windows 11 24H2

**Reply by milo-jo** [reply]
same issue

**Reply by ItsDevMacB** [reply]
Same

**Comment by citizenserious** [comment]
With Fedora 43 (KDE 6.5.3)
I am only seeing this since v3.0.1 last version tried v3.1.11 (rpm & AppImage)
<img width="935" height="666" alt="image" src="https://github.com/user-attachments/assets/0fe15fa9-195c-45e2-991e-80f3eab66020" />
Deactivating the firewall did not help.
I am connected to my network. Earlier versions (v2.x) did not have this specific issue.
Because I do not have any more information I didn't open an issue.
Does anyone know what's wrong here?
Thanks in advance (:

**Comment by pavlobu** [comment]
@citizenserious try this
https://github.com/pavlobu/deskreen?tab=readme-ov-file#linux

**Comment by pavlobu** [comment]
@citizenserious  also please share ifconfig command output. you can xx.xxx.xx the IPs and Mac Adresses - I don't need them. i just need to see your system network device aliases what they called to figure out what breaks it

FYI here is the code for this logic. if your network device is not there, then it is probably the cause
https://github.com/pavlobu/deskreen/blob/master/src/main/helpers/getMyLocalIpV4.ts

**Reply by citizenserious** [reply]
Hey @pavlobu , thank you for instant reply  (: 

```
 ifconfig | grep "inet "
        inet 127.0.0.1  netmask 255.0.0.0
        inet 192.168.x.x  netmask 255.255.255.0  broadcast 192.168.x.255
```
        
        
maybe this also helps:
```

deskreen-ce --ip 192.168.x.x
18:50:11.513 › /opt/Deskreen CE/resources/app.asar/out/main/index.jsc : skipping desktop capturer auto refresh on wayland session
18:50:11.519 › /opt/Deskreen CE/resources/app.asar/out/main/index.jsc : Deskreen CE signaling server is online at port 3131
18:50:11.520 › /opt/Deskreen CE/resources/app.asar/out/main/index.jsc : 🌐 Server available at http://localhost:3131
```

I also tried it with the AppImage (using Gear Lever)
<img width="548" height="53" alt="image" src="https://github.com/user-attachments/assets/21d914c7-2d23-4482-a44c-46b790dd2bdb" />

Unfortunately with the same behavior. 

Possible relevant info: I get notified about Deskreen updates.

**Reply by citizenserious** [reply]
Yes you are right! 
It has the name `wlo1` on my system, which is missing in `src/main/helpers/getMyLocalIpV4.ts`
Sorry I didn't understand that you wanted the full ifconfig at fist.


```
ifconfig
lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets **removed**
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets **removed**
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlo1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.x.x  netmask 255.255.255.0  broadcast 192.168.x.255
        inet6 **removed**
        inet6 **removed** 
        inet6 **removed** 
        inet6 **removed**
        inet6 **removed**
        ether **removed** txqueuelen 1000  (Ethernet)
        RX packets **removed**
        RX errors 0  dropped **removed**
        TX packets **removed**
        TX errors 0  dropped **removed**
```

**Comment by pavlobu** [comment]
@citizenserious thank you for your input!
this is useful and with that I can try to fix the bug.

By the way, do you know any reliable source to get all possible names of the network device interfaces aliases in entire zoo of Linux distros ?
So I can add them all in the code

**Reply by citizenserious** [reply]
Ahh, awesome. Thank you very much!

Normally I would not post this, but I think this actually could be a good answer, so why not share it if it could benefit this project.
I asked an AI assistant for help with your question about network interface naming, because I had no idea (I am not a programmer). The following recommendation **comes from the AI, not from me**.
If this AI-based input is not welcome, please excuse it — I simply try to help with the tools available to me (:

---

There is no reliable or complete source for *all* possible network interface names across the Linux ecosystem. The set of names is effectively unbounded because interfaces are named by several independent components:

* Kernel defaults (`eth0`, `wlan0`, etc.)
* systemd “predictable names” (`wlp2s0`, `enp3s0`, `wlo1`, `enx<MAC>`, ...)
* Distribution-specific udev rules
* Custom admin-defined names in `/etc/udev/rules.d/`
* Virtual interfaces (`docker0`, `virbr0`, `vethXXXX`, `tailscale0`, `ztXXXX`, `wg0`, ...)

Because of this, maintaining a hardcoded allowlist like:

```
wlan0, wlan1, wlpXsY, wlo1, enpXsY, eth0, eth1, ...
```

…will **always** miss some real-world cases.

---

## Recommended Approach

### 1. Enumerate all interfaces (Node.js)

Use `os.networkInterfaces()` and **filter out only obviously virtual or loopback prefixes**, such as:

```
lo, docker, veth, br-, virbr, tap, tun, wg, zt, vmnet
```

Then pick the first non-internal IPv4 address.

This approach is portable across distros and does not rely on naming conventions.

### 2. Optional: Determine the default-route IP

A more semantically correct approach is to detect the IP address of the interface used for the **default route**.

On Linux:

```bash
ip route get 1.1.1.1
```

The `src` field provides the actual outbound local IPv4.

Cross-platform npm packages like:

* `default-gateway`
* `internal-ip`

already implement this logic.

---

Maybe here comes the best part if it works :D

```
import os from 'os';

// Interfaces to skip completely (explicit names)
const IGNORE_INTERFACE_NAMES = [
  'lo',                           // Linux loopback
  'Loopback Pseudo-Interface 1',  // Windows loopback (common name)
];

// Interfaces to skip based on prefixes (virtual or container-related)
const IGNORE_INTERFACE_PREFIXES = [
  'lo',       // loopback
  'docker',   // Docker bridge
  'br-',      // Linux bridges
  'virbr',    // Libvirt/KVM virtual bridge
  'veth',     // Virtual ethernet peers
  'vmnet',    // VMware
  'zt',       // ZeroTier
  'tailscale',
  'wg',       // WireGuard
  'tun',      // TUN (virtual tunnel)
  'tap',      // TAP (virtual ethernet)
];

/**
 * Returns true if an IPv4 address is reasonable for LAN usage.
 * Excludes loopback (127.x.x.x) and link-local/APIPA (169.254.x.x).
 */
function isReasonableIPv4(address: string): boolean {
  if (!address) return false;
  if (address.startsWith('127.')) return false;      // loopback
  if (address.startsWith('169.254.')) return false;  // link-local (APIPA)
  return true;
}

/**
 * Returns true if the IPv4 address is part of private RFC1918 space.
 * This helps prefer true LAN IPs when multiple candidates exist.
 */
function isPrivateIPv4(address: string): boolean {
  const parts = address.split('.').map(Number);
  if (parts.length !== 4 || parts.some((n) => Number.isNaN(n))) return false;

  const [a, b] = parts;

  if (a === 10) return true;                        // 10.0.0.0/8
  if (a === 172 && b >= 16 && b <= 31) return true; // 172.16.0.0 – 172.31.255.255
  if (a === 192 && b === 168) return true;          // 192.168.0.0/16

  return false;
}

/**
 * Normalizes the network.family value to 'IPv4' | 'IPv6'.
 * Some environments historically used 4/6 instead of 'IPv4'/'IPv6'.
 */
function isIPv4Family(family: string | number): boolean {
  return family === 'IPv4' || family === 4;
}

/**
 * Attempts to determine the system's actual local IPv4 address
 * without relying on interface naming conventions.
 *
 * This approach:
 *  - Enumerates all interfaces from os.networkInterfaces()
 *  - Skips known virtual/loopback interfaces by name or prefix
 *  - Collects all valid, non-internal IPv4 addresses
 *  - Prefers private RFC1918 LAN addresses if available
 *
 * This is significantly more robust than hardcoding interface names
 * such as wlan0/wlp*/enp*/eth*, which vary across distros.
 */
export default function getMyLocalIpV4(): string | undefined {
  const networkInterfaces = os.networkInterfaces();

  const candidates: string[] = [];

  for (const [name, networks] of Object.entries(networkInterfaces)) {
    if (!networks || networks.length === 0) continue;

    // Skip obviously virtual or loopback interfaces
    if (IGNORE_INTERFACE_NAMES.includes(name)) continue;
    if (IGNORE_INTERFACE_PREFIXES.some((prefix) => name.startsWith(prefix))) {
      continue;
    }

    for (const network of networks) {
      if (
        isIPv4Family(network.family as string | number) &&
        !network.internal &&
        isReasonableIPv4(network.address)
      ) {
        candidates.push(network.address);
      }
    }
  }

  // No suitable IPv4 address found
  if (candidates.length === 0) {
    return undefined;
  }

  // Prefer private LAN addresses (typical home/office networks)
  const privateCandidate = candidates.find(isPrivateIPv4);
  if (privateCandidate) {
    return privateCandidate;
  }

  // Fallback: return the first reasonable IPv4 address
  return candidates[0];
}

```

**Reply by pavlobu** [reply]
thanks. yeah. AI is unreliable source. 
i will try to figure it out in the next release

**Reply by pavlobu** [reply]
@citizenserious you can try this solution
https://github.com/pavlobu/deskreen/issues/293
